### ISSUE TITLE
Initial mouse move issue

### ISSUE THREAD

**Issue by vinriviere** [aranym/aranym#25]
I run ARAnyM on Windows 10 64-bit. This is a problem with both ARAnyM 1.1.0 and latest snapshot. With SDL1 and SDL2 binaries. ARAnyM is running in a window.

- Run TOS/EmuTOS until desktop
- Move the mouse Left or Top : it doesn't move
- Move it Right or bottome : it moves well
- Move it Left or Top : now it works

It seems that the host mouse is not initialized at the center of the window, or something like that.
It worked well with old ARAnyM 0.9.16. This is a regression.

**Comment by vinriviere** [comment]
This really looks like a variant of issue #2. And indeed, I also noticed it with fVDI.

**Comment by joysfera** [comment]
Let me then close it, if it's already covered by #2.

**Comment by joysfera** [comment]
I'm reopening this until @vinriviere confirms that he was indeed using SDL2 version of ARAnyM (since only then it would be indeed variant of #2)

**Comment by vinriviere** [comment]
As I said initially, it equally fails with SDL1 or SDL2. Verified again with latest ARAnyM snapshot 126a6249 on Windows 10 64-bit and EmuTOS desktop.

**Comment by vinriviere** [comment]
This bug can be seen in this video (direct link to relevant part):
https://www.youtube.com/watch?v=GOkDuLmgWFo&t=551s

Specially, it is sad to see that any Windows user testing the official FreeMiNT setup will fatally hit this bug.

Historically, I used to move the mouse far bottom/right as a workaround. But when preparing the video, I found a quicker workaround: press middle then left mouse button. Definitely, something is wrongly initialized at startup in ARAnyM, since mouse ungrab/grab fixes the issue.

**Comment by th-otto** [comment]
Already posted part of this on the ML:

 if you don't move the mouse while starting aranym, then the initial mouse 
will be displayed at 64x64. That is some arbitrary default, which is set by 
the fVDI kernel, before any workstation is opened, and not corrected by the 
aranym driver (should actually not be needed, but apparently something is 
going wrong there).

- The issue does not occur when you set GrabMouse = No in the aranym config 
file

- the issue also seems to disappear, when you change the driver line in 
fvdi.sys to 

```
01r aranym.sys mode 1024x768x32@60 irq accelerate xxxxxxx
```
The important thing here is the irq keyword, which will cause fVDI to install 
a vbl handler to update the mouse position.

I've changed the default fvdi.sys in freemint for aranym accordingly (not yet contained in current snapshot, but should be there next time freemint is build)

And some more findings:

- on windows, the first call to retrieve SDL mouse position sometimes seem to return strange values when the mouse is initially grabbed. I've pushed a commit now that tries to work around this. 

- when the irq keyword is activated, fVDI bypasses TOS for most mouse actions, and amongst others, TOS will then not update the linea variables, so fVDI and TOS have their own idea where the mouse is located. I hope this will be fixed when the other mouse issues in fVDI are solved.

- the "invisible border" is caused by disagreement of the hosts/atari mouse position, and a similar effect that you have in Hatari. Actually, you could solve it the same way, by moving the mouse to all borders. Releasing/grabbing the mouse again as shown in the video is just a quicker way.

@vinriviere : please try the patch, whether that improves things. I'm still not totally convinced that this will solve it. If not, we'll have to wait for a fixed fVDI.

**Comment by vinriviere** [comment]
BTW, this still happens with plain EmuTOS (no fVDI) and the last binary snapshot from [there](https://github.com/aranym/aranym/issues/26#issuecomment-1214181899).
